packages/collector/src/agentConnection.js

```diff
@@ -158,7 +158,17 @@ exports.announceNodeCollector = function announceNodeCollector(cb) {
       payload.fd = String(socket._handle.fd);
 
       try {
-        payload.inode = fs.readlinkSync(pathUtil.join('/proc', String(process.pid), 'fd', payload.fd));
+        const linkPathPrefix = `${pathUtil.join('/proc', String(process.pid), 'fd')}/`;
+        const linkPath = pathUtil.join(linkPathPrefix, payload.fd);
+        payload.inode = fs.readlinkSync(linkPath);
+        if (typeof payload.inode === 'string' && payload.inode.indexOf(linkPathPrefix) === 0) {
```

Comment:
I was looking for the opposite test. Found it in /packages/collector/test/agentCommunication_test.js
But the test only runs on linux. Was wondering if you would like to cover the opposite case (return value is not a full path) in your bazel test

Reply:
Yes, sure, let's do that.

Done.